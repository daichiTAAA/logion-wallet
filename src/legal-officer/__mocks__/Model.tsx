import {
    fetchRecoveryInfo,
} from './ModelMock';

export {
    fetchRecoveryInfo,
};
