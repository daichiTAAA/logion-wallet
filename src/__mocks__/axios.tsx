function mockResponse(data: any) {
    return {
        data
    };
}

const axios = {

    get: (url: string) => mockResponse({}),
    put: (url: string) => mockResponse({}),
    post: (url: string) => mockResponse({}),
};
export default axios;

export function mockPut(whenUrl: string, thenData: any, thenCall?: () => void) {
    axios.put = (url: string): any => {
        if(url === whenUrl) {
            if(thenCall !== undefined) {
                thenCall();
            }
            return mockResponse(thenData);
        }
    };
}

export function mockPost(whenUrl: string, thenData: any, thenCall?: () => void) {
    axios.post = (url: string): any => {
        if(url === whenUrl) {
            if(thenCall !== undefined) {
                thenCall();
            }
            return mockResponse(thenData);
        }
    };
}
