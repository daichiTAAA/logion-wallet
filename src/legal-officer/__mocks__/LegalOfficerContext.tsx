export let pendingTokenizationRequests: any[] | null = null;

export let rejectRequest = () => {

};

export let acceptedTokenizationRequests: any[] | null = null;

export let rejectedTokenizationRequests: any[] | null = null;

export let refreshRequests = () => {

};

export let pendingProtectionRequests: any[] | null = null;

export let acceptedProtectionRequests: any[] | null = null;

export let protectionRequestsHistory: any[] | null = null;

export function useLegalOfficerContext() {
    return {
        pendingTokenizationRequests,
        rejectRequest,
        rejectedTokenizationRequests,
        acceptedTokenizationRequests,
        refreshRequests,
        pendingProtectionRequests,
        acceptedProtectionRequests,
        protectionRequestsHistory,
    };
}

export function setPendingRequests(requests: any[]) {
    pendingTokenizationRequests = requests;
}

export function setRejectRequest(callback: any) {
    rejectRequest = callback;
}

export function setRejectedRequests(requests: any[]) {
    rejectedTokenizationRequests = requests;
}

export function setAcceptedRequests(requests: any[]) {
    acceptedTokenizationRequests = requests;
}

export function setRefreshRequests(callback: any) {
    refreshRequests = callback;
}

export function setPendingProtectionRequests(requests: any[]) {
    pendingProtectionRequests = requests;
}

export function setAcceptedProtectionRequests(requests: any[]) {
    acceptedProtectionRequests = requests;
}

export function setProtectionRequestsHistory(requests: any[]) {
    protectionRequestsHistory = requests;
}
