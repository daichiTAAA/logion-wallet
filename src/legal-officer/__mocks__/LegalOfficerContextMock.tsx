export let pendingTokenizationRequests: any[] | null = null;

export let rejectRequest = () => {

};

export let acceptedTokenizationRequests: any[] | null = null;

export let rejectedTokenizationRequests: any[] | null = null;

export let refreshRequests = jest.fn();

export let pendingProtectionRequests: any[] | null = null;

export let activatedProtectionRequests: any[] | null = null;

export let protectionRequestsHistory: any[] | null = null;

export let recoveryRequestsHistory: any[] | null = null;

export function setPendingRequests(requests: any[]) {
    pendingTokenizationRequests = requests;
}

export function setRejectRequest(callback: any) {
    rejectRequest = callback;
}

export function setRejectedRequests(requests: any[]) {
    rejectedTokenizationRequests = requests;
}

export function setAcceptedRequests(requests: any[]) {
    acceptedTokenizationRequests = requests;
}

export function setRefreshRequests(callback: any) {
    refreshRequests = callback;
}

export function setPendingProtectionRequests(requests: any[]) {
    pendingProtectionRequests = requests;
}

export function setActivatedProtectionRequests(requests: any[]) {
    activatedProtectionRequests = requests;
}

export function setProtectionRequestsHistory(requests: any[]) {
    protectionRequestsHistory = requests;
}

export function setRecoveryRequestsHistory(requests: any[]) {
    recoveryRequestsHistory = requests;
}
