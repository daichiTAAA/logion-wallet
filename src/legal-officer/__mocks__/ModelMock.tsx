export const DEFAULT_LEGAL_OFFICER = "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"; // Alice
export const ANOTHER_LEGAL_OFFICER = "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"; // Bob

export let rejectRequest = jest.fn();

export let acceptRequest = jest.fn();

export function setAcceptRequest(mockFn: any) {
    acceptRequest = mockFn;
}

export let setAssetDescription = jest.fn();

export let rejectProtectionRequest = jest.fn();

export function setRejectProtectionRequest(mockFn: any) {
    rejectProtectionRequest = mockFn;
}

export let acceptProtectionRequest = jest.fn();

export function setAcceptProtectionRequest(mockFn: any) {
    acceptProtectionRequest = mockFn;
}

export function decision(legalOfficerAddress: string, decisions: any[]) {
    return decisions[0];
}

export let fetchRecoveryInfo = jest.fn();

export function setFetchRecoveryInfo(mockFn: any) {
    fetchRecoveryInfo = mockFn;
}
