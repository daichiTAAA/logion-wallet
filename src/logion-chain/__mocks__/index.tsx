import { Extension } from '../Keys';

const LogionChainContextProvider = (props: any) => null;

let context = {
    apiState: 'CONNECT_INIT',
    injectedAccounts: []
};

export function setContextMock(value: any) {
    context = value;
}

export function useLogionChain() {
    return context;
}

let extensionAvailable = false;

export function setExtensionAvailable(value: boolean) {
    extensionAvailable = value;
}

export function isExtensionAvailable() {
    return extensionAvailable;
}

let recommendedExtensionValue: Extension | null = null;

export function setRecommendedExtension(value: Extension | null) {
    recommendedExtensionValue = value;
}

export function recommendedExtension(): Extension | null {
    return recommendedExtensionValue;
}

export {
    LogionChainContextProvider,
};
