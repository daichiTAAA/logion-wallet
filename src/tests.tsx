import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
