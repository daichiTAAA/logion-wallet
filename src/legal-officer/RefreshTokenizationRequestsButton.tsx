import React from 'react';
import Button from 'react-bootstrap/Button';

import { useLegalOfficerContext } from './LegalOfficerContext';

export default function RefreshTokenizationRequestsButton() {
    const { refreshRequests } = useLegalOfficerContext();
    if(refreshRequests === null) {
        return null;
    } else {
        return <Button onClick={refreshRequests}>Refresh requests</Button>;
    }
}
