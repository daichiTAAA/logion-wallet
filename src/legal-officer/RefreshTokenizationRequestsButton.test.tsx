jest.mock('./LegalOfficerContext');

import RefreshTokenizationRequestsButton from './RefreshTokenizationRequestsButton';
import { shallowRender } from '../tests';
import { setRefreshRequests } from './__mocks__/LegalOfficerContextMock';

test("Renders null with no data", () => {
    setRefreshRequests(null);
    const tree = shallowRender(<RefreshTokenizationRequestsButton />);
    expect(tree).toMatchSnapshot();
});

test("Renders with refreshRequests", () => {
    const refreshRequests = jest.fn();
    setRefreshRequests(refreshRequests);
    const tree = shallowRender(<RefreshTokenizationRequestsButton />);
    expect(tree).toMatchSnapshot();
});
