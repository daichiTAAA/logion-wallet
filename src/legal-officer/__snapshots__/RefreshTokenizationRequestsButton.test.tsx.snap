// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders null with no data 1`] = `null`;

exports[`Renders with refreshRequests 1`] = `
<Button
  active={false}
  disabled={false}
  onClick={[MockFunction]}
  variant="primary"
>
  Refresh requests
</Button>
`;
